src/snapshot/metadata.rs

```diff
@@ -125,11 +135,29 @@ impl InstanceContext {
     }
 }
 
+// serve_session always passes an option from the config file, but tests want it to be explict
+#[cfg(test)]
+impl From<ScriptContextType> for InstanceContext {
+    fn from(script_type: ScriptContextType) -> Self {
+        Self {
+            script_type,
+            ..Self::new()
+        }
+    }
+}
+
+impl From<Option<ScriptContextType>> for InstanceContext {
+    fn from(script_type: Option<ScriptContextType>) -> Self {
+        Self {
+            script_type: script_type.unwrap_or_default(),
+            ..Self::new()
+        }
+    }
+}
+
```

Comment:
I wasn't sure if I should keep this in for the tests. Let me know your thoughts.